contracts/src/libraries/Challenge.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.17;
+
+import "./Bytes.sol";
+
+/**
+ * @title Library of challenge generation algorithm
+ * @author Seiya Kobayashi
+ */
+library Challenge {
+    /**
+     * @notice Generate a random challenge.
+     * @dev Generate a challenge of random 32 bytes (bytes32) of the given difficulty.
+     * @param _difficulty Difficulty of challenge
+     * @return challenge Generated challenge
+     */
+    function generateChallenge(
```

Comment:
Only the last N digits of length of `difficulty` is randomly generated.

e.g.)

- `difficulty` == 4 (bits)
```sh
0x000000000000000000000000000000000000000000000000000000000000f2ac
```

- `difficulty` == 7 (bits)
```sh
0x00000000000000000000000000000000000000000000000000000000097cf2ac
```